### PR TITLE
SC194961 (1 of 2) download gisaid submission template

### PR DESCRIPTION
**Pull request by ehoops-cz**
### Summary:
- **What:** `Frontend for downloading gisaid template.  This PR also includes some minor accessibility improvements.`
- **Ticket:** [sc194961](https://app.shortcut.com/genepi/story/194961/as-a-user-i-want-to-select-to-download-my-gisaid-submission-template-files)
- **Env:** `none`

### Demos:
Screen Shot 2022-07-29 at 8 42 02 AM

Screen Shot 2022-07-29 at 8 42 09 AM

### Notes:
There are still TODOs for:
* Connecting to the backend API when it's ready
* Adding the href for the Learn More link

I will create a second PR for these ^

### Checklist:
- [ ] I merged latest `<base branch>`
- [ ] I manually verified the change
- [ ] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx [hunk 1]
```diff
@@ -173,43 +183,69 @@ const DownloadModal = ({
...
-                        Download multiple consensus genomes in a single,
-                        concatenated file
-                      </DownloadTypeInfo>
-                    </CheckBoxInfo>
-                  </CheckBoxWrapper>
-                </StyledSpan>
+                <StyledFileTypeItem
+                  isDisabled={isFastaDisabled}
+                  isSelected={isFastaSelected}
+                >
```
Comment (ehoops-cz) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:189]: added `isDisabled` and `isSelected` as props to the styled component and moved the background color styling there.

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx [hunk 2]
```diff
@@ -173,43 +183,69 @@ const DownloadModal = ({
...
+                  <CheckBoxInfo>
+                    <StyledCheckbox
+                      disabled={isFastaDisabled}
+                      id="download-fasta-checkbox"
+                      onChange={handleFastaClick}
+                      stage={isFastaSelected ? "checked" : "unchecked"}
```
Comment (ehoops-cz) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:195]: Updated to align with new sds syntax (🤦‍♀️ trying to block out adding that to sds without versioning the release, sorry!)

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx [hunk 3]
```diff
@@ -173,43 +183,69 @@ const DownloadModal = ({
...
+                    />
+                  </CheckBoxInfo>
+                  <CheckboxLabel htmlFor="download-fasta-checkbox">
```
Comment (ehoops-cz) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:198]: Accessibility: Checkboxes need to have labels.  In our case, we had a visible label, but the screen reader didn't know to connect the two.  Now we have a semantic `label` element and the `htmlFor` field connects it to the checkbox element above.  Now the screen reader announces the label 🎉

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx [hunk 4]
```diff
@@ -173,43 +183,69 @@ const DownloadModal = ({
...
                   <DownloadTypeInfo>
                     Sample metadata including Private and Public IDs, Collection
                     Date, Sequencing Date, Lineage, GISAID Status, and ISL
                     Accession #.
                   </DownloadTypeInfo>
-                </CheckBoxInfo>
-              </CheckBoxWrapper>
+                </CheckboxLabel>
+              </StyledFileTypeItem>
+              {isPrepFilesFlagOn && (
```
Comment (ehoops-cz) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:226]: Using feature flag here

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/style.ts [hunk 5]
```diff
@@ -38,43 +39,52 @@ export const Title = styled.span`
   }}
 `;
 
+const checkBoxInfoSpacing = (props: CommonThemeProps) => {
+  const spaces = getSpaces(props);
+  return `
+    position: inline-block;
+    float: left;
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/style.ts:46]: you can feel free to change `float`s to `flex`s if you feel like it! (but no pressure 💪)

**Review by mmmmmmaya — APPROVED** [review]

**Pull request merged**